Common/src/geometry/CGeometry.cpp

```diff
@@ -969,14 +976,22 @@ void CGeometry::PreprocessPeriodicComms(CGeometry* geometry, CConfig* config) {
              index on the matching periodic point and the periodic marker
              index to be communicated to the recv rank. ---*/
 
-            if (iRank == Neighbors_PeriodicSend[iSend]) {
-              Local_Point_PeriodicSend[ii] = iPoint;
-              Local_Marker_PeriodicSend[ii] = static_cast<unsigned long>(iMarker);
-              jj = ii * nPackets;
-              idSend[jj] = geometry->vertex[iMarker][iVertex]->GetDonorPoint();
-              jj++;
-              idSend[jj] = static_cast<unsigned long>(iPeriodic);
-              ii++;
+            if (iRank == destRank) {
+              /*--- Check if we have already added this (point, marker) pair
+               for this destination rank. ---*/
+              auto pointMarkerPair = std::make_pair(iPoint, static_cast<unsigned long>(iMarker));
+              if (Points_Send_All[destRank].find(pointMarkerPair) == Points_Send_All[destRank].end()) {
+                Local_Point_PeriodicSend[ii] = iPoint;
+                Local_Marker_PeriodicSend[ii] = static_cast<unsigned long>(iMarker);
+                jj = ii * nPackets;
+                idSend[jj] = geometry->vertex[iMarker][iVertex]->GetDonorPoint();
+                jj++;
+                idSend[jj] = static_cast<unsigned long>(iPeriodic);
+                ii++;
+
+                /*--- Store the (point, marker) pair in the set for the destination rank. ---*/
+                Points_Send_All[destRank].insert(pointMarkerPair);
```

Comment:
You can always call insert and use the result of insert to determine whether this is the first time the pair is found. This saves the double work of calling find and then insert.